system/Helpers/form_helper.php

```diff
@@ -0,0 +1,954 @@
+        {
+            $content = $data['content'];
+            unset($data['content']); // content is not an attribute
+        }
+        
+        return '<button '._parse_form_attributes($data, $defaults)._attributes_to_string($extra).'>'
+                .$content
+                ."</button>\n";
+    }
+}
+
+// ------------------------------------------------------------------------
+
+if ( ! function_exists('form_label'))
+{
+    /**
+     * Form Label Tag
+     *
+     * @param	string	The text to appear onscreen
+     * @param	string	The id the label applies to
+     * @param	array	Additional attributes
+     * @return	string
+     */
+    function form_label(string $label_text = '', string $id = '', array $attributes = []): string
+    {
+        $label = '<label';
+        
+        if ($id !== '')
+        {
+            $label .= ' for="'.$id.'"';
+        }
+        
+        if (is_array($attributes) && count($attributes) > 0)
+        {
+            foreach ($attributes as $key => $val)
+            {
+                $label .= ' '.$key.'="'.$val.'"';
+            }
+        }
+        
+        return $label.'>'.$label_text.'</label>';
+    }
+}
+
+// ------------------------------------------------------------------------
+
+if ( ! function_exists('form_datalist'))
+{
+    /**
+     * Datalist
+     * 
+     * The <datalist> element specifies a list of pre-defined options for an <input> element.
+     * Users will see a drop-down list of pre-defined options as they input data.
+     * The list attribute of the <input> element, must refer to the id attribute of the <datalist> element.
+     */
+    function form_datalist($name, $value, $options)
+    {
+        $data = [
+             'type' => 'text',
+             'name' => $name,
+             'list' => $name.'_list',
+             'value' => $value
+        ];
+        
+        $out = form_input($data)."\n";
+        $out .= "<datalist id='".$name.'_list'."'>";
+        
+        foreach ($options as $option)
+        {
+            $out .= "<option value='$option'>"."\n";
+        }
+        
+        $out .= "</datalist>"."\n";
+        
+        return $out;
+    }
+}
+
+// ------------------------------------------------------------------------
+
+if ( ! function_exists('form_keygen'))
+{
+    function form_keygen()
+    {
+        /* The purpose of the <keygen> element is to provide a secure way to authenticate users.
+         The <keygen> element specifies a key-pair generator field in a form.
+         When the form is submitted, two keys are generated, one private and one public.
+         The private key is stored locally, and the public key is sent to the server.
+         The public key could be used to generate a client certificate to authenticate the user in the future. */
+    }
+}
+
+// ------------------------------------------------------------------------
+
+if ( ! function_exists('form_output'))
+{
+    function form_output()
```

Comment:
This should be possible as a simple field. While other fields need to be attached to it, usually through JS, the output field itself seems to be a standalone field if I understand this one correctly.